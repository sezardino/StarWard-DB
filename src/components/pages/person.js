import React, {PureComponent} from 'react';
import Row from '../row';
import {PersonList, PersonDetails} from '../app-components';

export default class PersonPage extends PureComponent {
	state = {
		selectedItem: null,
	};

	itemClickHandler = (selectedItem) => {
		this.setState({selectedItem});
	};

	render() {
		const {selectedItem} = this.state;
		return (
			<Row
				left={
					<PersonList renderItem={({name}) => `${name}`} onListItemClick={this.itemClickHandler} />
				}
				right={<PersonDetails dataId={selectedItem} />}
			/>
		);
	}
}
