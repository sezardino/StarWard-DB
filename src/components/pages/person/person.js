import React from 'react';
import {withRouter} from 'react-router-dom';
import Row from '../../row';
import {PersonList, PersonDetails} from '../../app-components';

const PersonPage = ({history, match}) => {
	let id = match.params.id ? match.params.id : null;

	return (
		<Row
			left={
				<PersonList renderItem={({name}) => `${name}`} onListItemClick={(id) => history.push(id)} />
			}
			right={<PersonDetails dataId={id} />}
		/>
	);
};

export {PersonPage};
export default withRouter(PersonPage);
