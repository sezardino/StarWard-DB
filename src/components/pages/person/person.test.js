import React from 'react';
import {shallow} from 'enzyme';

import {PersonPage} from './person';

describe('Component PersonPage', () => {
	const mockMatch = {params: {id: 'Test id'}};
	const component = shallow(<PersonPage match={mockMatch} />);
	it('Component rendered correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});
});
