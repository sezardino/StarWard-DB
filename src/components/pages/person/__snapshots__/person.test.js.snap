// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PersonPage Component rendered correctly 1`] = `"<Row left={{...}} right={{...}} />"`;
